core/src/test/scala/unit/kafka/server/AbstractFetcherThreadTest.scala

```diff
@@ -144,6 +144,44 @@ class AbstractFetcherThreadTest {
     assertEquals(2L, replicaState.highWatermark)
   }
 
+  @Test
+  def testDelay(): Unit = {
+    val partition = new TopicPartition("topic", 0)
+
+    class ErrorMockFetcherThread(fetchBackOffMs: Int)
+      extends MockFetcherThread(fetchBackOffMs =  fetchBackOffMs) {
+
+      override def fetchFromLeader(fetchRequest: FetchRequest.Builder): Map[TopicPartition, FetchData] = {
+         throw new UnknownTopicIdException("Topic ID was unknown as expected for this test")
+      }
+    }
+    val fetcher = new ErrorMockFetcherThread(fetchBackOffMs = 1000)
```

Comment:
nit: can you create a `val` for `fetchBackoffMs`? Then we can use it below as well.

Also, could we use a lower value for the backoff?

Reply:
Sure. I just went with the default value. Would 500 still be too high? In my testing, the non-delayed call usually took 30ms on my local machine, so we have quite a ways to go before we get close.